src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
+const FULL_HEALTH = 100;
+const BITE_DAMAGE = 50;
+
 class Animal {
-  // write your code here
+  i;
+  static alive = [];
+
+  constructor(name, health = FULL_HEALTH) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = FULL_HEALTH) {
+    super(name, health);
+    this.hidden = false;
```

Comment:
[CODE DUPLICATION] - The `hidden` property is being initialized to `false` again in the `Herbivore` constructor, which is redundant as this has already been done in the `Animal` class constructor. Since the `hidden` property should only belong to `Herbivore`, remove it from the `Animal` class and keep the initialization here in the `Herbivore` constructor.